### PR TITLE
Support HTTP Proxy from environment

### PR DESCRIPTION
**Pull request by hangscer8**
Add the support of http prxoy to `NewClientWithEnvProxy`.

Relates to: #2363.

### CODE REVIEW

**Comment by gmlewis** [comment]
@hangscer8 - I'm concerned that this change is going to affect current users of this repo who are not expecting this behavior to be the default.

Could we instead add a new `NewClientWithEnvProxy` or something like that?

**Comment by hangscer8** [comment]
> @hangscer8 - I'm concerned that this change is going to affect current users of this repo who are not expecting this behavior to be the default.
> 
> Could we instead add a new `NewClientWithEnvProxy` or something like that?

OK, I will fix that. @gmlewis

**Review by gmlewis — APPROVED** [review]
Thank you, @hangscer8 !
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
> LGTM!

Thank you, @valbeat !
Merging.

**Pull request merged**
